### DISCUSSION TITLE
The website

### DISCUSSION THREAD

**Discussion by Vivraan** [Godot-Languages-Support/godot-lang-support#63] [Ideas]
This is an offshoot of #61 specifically for discussing what to do with the website version of Godot Language Support.

Currently, some of the features discussed are:

1. Generating the page based on data pulled from YAML that may benefit from GitHub actions automation.
2. A way to detect the freshness/staleness of a language project repository, possibly through GitHub actions.

Please direct your discussion about the website here.

**Comment by Croydon** [comment]
Just my two quick cents: I think it would be better to use individual issues once there is an initial website done and `A way to detect the freshness/staleness of a language project repository, possibly through GitHub actions.` feels very advanced, so I wouldn't concentrate on that initially.

A Yaml file, that is getting transformed into a table, that can be sorted by individual columns and filtered by search would be awesome. Additionally, a checkbox that hides/shows unmaintained projects. That said, any sorting, filtering, and hiding could also be later steps. Make it quick and dirty first and improve from there.

**Reply by ShalokShalom** [reply]
Yeah, I was kinda thinking we are using a static homepage, and that would mean its in markdown, no? 

Other solutions are obviously also possible.

**Reply by Vivraan** [reply]
I agree completely: we must focus on priority objectives first. Pulling data from YAML seems to be the first priority.

**Comment by ShalokShalom** [comment]
I was just looking. Dynamic tables, who are sortable and that kind of stuff, are not possible with markdown.

As this seemingly excludes static webpages, I ask myself what we would want to use instead.

I know that wasm files can be hosted on Github pages, Netlify and so on as well.

That would bring a funny idea to my mind: Make it in Godot. 🙂

How about that?

**Reply by Vivraan** [reply]
Are WASM builds possible with Godot 4? I still think it's possible with 3, no?

In any case I need to look into this YAML based thing and see if we can do anything with it. The promise of hosting multiple config files which need only be updated from is amazing since it promises config files to be readily available for pull requests.

I was considering if it was more HTML/JS based – markdown seems less featured for this sort of task? I still need to understand what is possible.

**Reply by ShalokShalom** [reply]
I am not quite sure we are all on the same page, what means "that YAML based thing" 

I also dont quite understand the benefit you see with it.

**Reply by ShalokShalom** [reply]
Web export works for GDScript and C# at least.

The ability to create games with it seems not perfect, but totally enough for a webpage.

The nice thing is, we could even load Godot itself - with language examples.

So example projects in Zig, Go and so on.

So people get an actual impression, how the language utilizes the API and so on.

Thats something the community could contribute (and probably way too much again 😅)

https://godotengine.org/article/progress-report-web-export-in-4-3/

**Reply by Vivraan** [reply]
Haha, that might work for v3.0 on this thing 😂

We must stick to something easy for this iteration. I think creating a full fledged webpage with playable implementations will require solid community effort; we can't do that with our current size.

**Reply by Croydon** [reply]
A static webpage just means that it does not perform some logic on the server side that decides what the users see. JavaScript can run in the browser and it is more than enough for some sorting and filtering of a table. Using Godot for the website is overkill.

**Reply by Vivraan** [reply]
> I am not quite sure we are all on the same page, what means "that YAML based thing"
> 
> I also dont quite understand the benefit you see with it.

I don't quite know as well xD

@winston-yallow could elaborate on it, but I need to do my reading.

YAML implies using GitHub actions to automate content displayed on the page, in which case the YAML files would contain the site data, perhaps per project. We could also have a monolithic file.

The benefit I saw in this apparent setup is allowing people to contribute YAML files for their own projects, or co-author ones that we may set up ourselves.

**Reply by Vivraan** [reply]
> A static webpage just means that it does not perform some logic on the server side that decides what the users see. JavaScript can run in the browser and it is more than enough for some sorting and filtering of a table. Using Godot for the website is overkill.

Yeah, I don't see any problem with a static webpage that pulls content from _somewhere_ as well.

I suppose we all have different ideas about what this web resource should look like; let's discuss and perhaps draw a consensus? 🫡😁

**Reply by ShalokShalom** [reply]
I am fine with everything 🙂

**Comment by Vivraan** [comment]
Hello hello!
Just want to check how everyone's doing? Would it help if we had an online meeting to discuss initial design stuff?
